ql/src/java/org/apache/hadoop/hive/ql/exec/FileSinkOperator.java

```diff
@@ -736,6 +740,21 @@ protected void initializeOp(Configuration hconf) throws HiveException {
     }
   }
 
+  private boolean skipPartitionCheck() {
+    return Optional.ofNullable(conf).map(FileSinkDesc::getTableInfo)
+        .map(TableDesc::getProperties)
+        .map(props -> props.getProperty(hive_metastoreConstants.META_TABLE_STORAGE))
+        .map(handler -> {
+          try {
+            return HiveUtils.getStorageHandler(hconf, handler);
+          } catch (HiveException e) {
+            return null;
+          }
+        })
+        .map(HiveStorageHandler::alwaysUnpartitioned)
```

Comment:
No, it shouldn't. If we return null in any of the map functions in Optional, then it will fall back to the `orElse` part (i.e. subsequent map functions won't fail)